graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -42,10 +42,15 @@
 
     step V-4. Recurse into any vertex field children of the current AST node:
               (see _compile_vertex_ast())
-        - before recursing into each vertex, process any @optional directive
-          present on the child AST node;
-        - after returning from each vertex, return to the marked query location using
-          the appropriate manner, depending on whether @optional was present on the child or not.
+          - before recursing into each vertex:
```

Comment:
All indentation here is two spaces deeper than the corresponding bullet points in V-3.